_posts/2023-07-07-long-running-operation-and-component-template.md

```diff
@@ -0,0 +1,116 @@
+
+<img src="/assets/media/blog-images/2023-07-07-long-running-operation-and-component-template/support-adhoc-notification-config.png" alt="Support adhoc notification config"/>{: .img-fluid }
+
+For more information, see [Configuring notification settings for an individual operation](https://opensearch.org/docs/latest/dashboards/im-dashboards/notifications/#configuring-notification-settings-for-an-individual-operation).
+
+### Configuring notifications through the API
+
+You can use the `lron` API endpoint to configure notification settings for a task. For example, the following query sets up notifications when a reindex task fails:
+
+```json
+POST /_plugins/_im/lron
+{
+  "lron_config": {
+      "task_id":"dQlcQ0hQS2mwF-AQ7icCMw:12354",
+      "action_name":"indices:data/write/reindex",
+      "lron_condition": {
+        "success": false,
+        "failure": true
+      },
+      "channels":[
+          {"id":"channel1"},
+          {"id":"channel2"}
+      ]
+  }
+}
+```
+
+The `lron` API lets you create, retrieve, update, and delete notification settings. To learn more about these operations, see [Notification settings](https://opensearch.org/docs/latest/im-plugin/notifications-settings/).
+
+### Setting up fine-grained access control with notification permissions
+
+To limit access to notification setup to certain users, notifications are integrated with the Security plugin, which provides the following fine-grained permissions:
+
+- `cluster:admin/opensearch/controlcenter/lron/get`: The user has permission to `view` the notification configs for long-running operations.
+- `cluster:admin/opensearch/controlcenter/lron/write`: The user has permission to `add or update` the notification configs for long-running operations.
+- `cluster:admin/opensearch/controlcenter/lron/delete`: The user has permission to `delete` the notification configs for long-running operations.
+
+If a user lacks permissions to view notification settings, the user is notified to request permissions.
+
+<img src="/assets/media/blog-images/2023-07-07-long-running-operation-and-component-template/security-enabled.png" alt="Security enabled"/>{: .img-fluid }
+
+## Component templates
+
+Component templates let you create a single index pattern that matches multiple indexes. You can combine multiple component templates to create an index template.
+
+### Creating component templates
+
+When you create index templates through the API, it may be difficult to tell what the index template will be like after merging all the associated component templates with their own configurations. Now when you create or update an index template through the UI, you can preview the template in OpenSearch Dashboards.
```

Comment:
```suggestion
When you create index templates through the API, it may be difficult to determine what the index template will be like after merging all the associated component templates with their own configurations. Now when you create or update an index template through the UI, you can preview the template in OpenSearch Dashboards.
```